cmd/config.go

```diff
@@ -0,0 +1,175 @@
+		allConfig := make(map[string][]config.ConfigSnippet)
+
+		renderErr := 0
+
+		for _, node := range c.Nodes {
+			kind := node.Labels["clab-node-kind"]
+			err = config.LoadTemplate(kind, templatePath)
+			if err != nil {
+				return err
+			}
+
+			res, err := config.RenderNode(node)
+			if err != nil {
+				log.Errorln(err)
+				renderErr += 1
+				continue
+			}
+			allConfig[node.LongName] = append(allConfig[node.LongName], res...)
+
+		}
+
+		for lIdx, link := range c.Links {
+
+			res, err := config.RenderLink(link)
+			if err != nil {
+				log.Errorf("%d. %s\n", lIdx, err)
+				renderErr += 1
+				continue
+			}
+			for _, rr := range res {
+				allConfig[rr.TargetNode.LongName] = append(allConfig[rr.TargetNode.LongName], rr)
+			}
+
+		}
+
+		if renderErr > 0 {
+			return fmt.Errorf("%d render warnings", renderErr)
+		}
+
+		if printLines > 0 {
+			// Debug log all config to be deployed
+			for _, v := range allConfig {
+				for _, r := range v {
+					r.Print(printLines)
+				}
+			}
+			return nil
+		}
+
+		var wg sync.WaitGroup
+		wg.Add(len(allConfig))
+		for _, cs_ := range allConfig {
+			deploy1 := func(cs []config.ConfigSnippet) {
+				defer wg.Done()
+
+				var transport config.Transport
+
+				ct, ok := cs[0].TargetNode.Labels["config.transport"]
+				if !ok {
+					ct = "ssh"
+				}
+
+				if ct == "ssh" {
+					transport, _ = newSSHTransport(cs[0].TargetNode)
+					if err != nil {
+						log.Errorf("%s: %s", kind, err)
+					}
+				} else if ct == "grpc" {
+					// newGRPCTransport
+				} else {
+					log.Errorf("Unknown transport: %s", ct)
+					return
+				}
+
+				err := config.WriteConfig(transport, cs)
+				if err != nil {
+					log.Errorf("%s\n", err)
+				}
+			}
+
+			// On debug this will not be executed concurrently
+			if log.IsLevelEnabled(log.DebugLevel) {
+				deploy1(cs_)
+			} else {
+				go deploy1(cs_)
+			}
+		}
+		wg.Wait()
+
+		return nil
+	},
+}
+
+func newSSHTransport(node *clab.Node) (*config.SshTransport, error) {
+	switch node.Kind {
+	case "vr-sros", "srl":
+		c := &config.SshTransport{}
+		c.SshConfig = &ssh.ClientConfig{}
+		config.SshConfigWithUserNamePassword(
+			c.SshConfig,
+			clab.DefaultCredentials[node.Kind][0],
+			clab.DefaultCredentials[node.Kind][1])
+
+		switch node.Kind {
+		case "vr-sros":
+			c.K = &config.VrSrosSshKind{}
+		case "srl":
+			c.K = &config.SrlSshKind{}
+		}
+		return c, nil
+	}
+	return nil, fmt.Errorf("no tranport implemented for kind: %s", kind)
+}
+
+func init() {
+	rootCmd.AddCommand(configCmd)
+	configCmd.Flags().StringVarP(&templatePath, "path", "", "", "specify template path")
+	configCmd.MarkFlagDirname("path")
+	configCmd.Flags().StringVarP(&config.TemplateOverride, "templates", "", "", "specify a list of template to apply")
+	configCmd.Flags().IntVarP(&printLines, "print-only", "p", 0, "print config, don't send it. Restricted to n lines")
+	configCmd.Flags().BoolVarP(&config.LoginMessages, "login-message", "", false, "show the SSH login message")
```

Comment:
if this is only used as a debug thingy, maybe remove the flag and use debug level instead?